aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,273 @@
+            "calctype",
+            valid_type=Str,
+            default=lambda: Str("singlepoint"),
+            help="calculation type (single point or geom opt)",
+        )
+        spec.input(
+            "architecture",
+            valid_type=Str,
+            default=lambda: Str("mace_mp"),
+            help="Architecture to use for calculation, defaults to mace_mp",
+        )
+        spec.input(
+            "model",
+            valid_type=ModelData,
+            required=False,
+            help="mlip model used for calculation",
+        )
+        spec.input("structure", valid_type=StructureData, help="The input structure.")
+        spec.input("precision", valid_type=Str, help="Precision level for calculation")
+        spec.input(
+            "device",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str("cpu"),
+            help="Device on which to run calculation (cpu, cuda or mps)",
+        )
+
+        spec.input(
+            "xyz_output_name",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str(cls._XYZ_OUTPUT),
+            help="Name of the xyz output file",
+        )
+
+        spec.input(
+            "log_filename",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str(cls._LOG_FILE),
+            help="Name of the log output file",
+        )
+        spec.input(
+            "metadata.options.output_filename",
+            valid_type=str,
+            default=cls._DEFAULT_OUTPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.input_filename",
+            valid_type=str,
+            default=cls._DEFAULT_INPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.scheduler_stdout",
+            valid_type=str,
+            default="_scheduler-stdout.txt",
+            help="Filename to which the content of stdout of the scheduler is written.",
+        )
+        spec.inputs["metadata"]["options"]["parser_name"].default = "janus.parser"
+        spec.inputs.validator = cls.validate_inputs
+        # Define outputs. The default is a dictionary with the content of the xyz file
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
+        )
+        print("defining outputs")
+        spec.output("std_output", valid_type=SinglefileData)
+        spec.output("log_output", valid_type=SinglefileData)
+        spec.output("xyz_output", valid_type=SinglefileData)
+        print("defining outputnode")
+        spec.default_output_node = "results_dict"
+
+        # Exit codes
+
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+        spec.exit_code(
+            306,
+            "ERROR_EMPTY_OUTPUT_FILES",
+            message="File.xyz is empty",
```

Comment:
Is this file always called "File.xyz"? If not this is an unhelpful error message.